Formula/apache-httpd.rb

```diff
@@ -0,0 +1,128 @@
+class ApacheHttpd < Formula
+  desc "Apache HTTP server"
+  homepage "https://httpd.apache.org/"
+  url "https://www.apache.org/dyn/closer.cgi?path=httpd/httpd-2.4.27.tar.bz2"
+  sha256 "71fcc128238a690515bd8174d5330a5309161ef314a326ae45c7c15ed139c13a"
+
+  bottle do
+    sha256 "d35e67ae745053cda23273a520ead6bde506ec8082057edcd3185c9e36eae483" => :sierra
+    sha256 "4ea7056d6c08f9cf49507869aa8c78e3522c601d87b3347d31561ba05b33d41c" => :el_capitan
+    sha256 "e6ae5c4d38b40bcf3d85877ac117a1ea048de774edec3c2b0ff714bc01a6d40a" => :yosemite
+  end
+
+  deprecated_option "with-http2" => "with-nghttp2"
+
+  depends_on "apr"
+  depends_on "apr-util"
+  depends_on "openssl"
+  depends_on "pcre"
+
+  depends_on "nghttp2" => :optional
+
+  def install
+    # point config files to opt_prefix instead of the version-specific prefix
+    inreplace "Makefile.in",
+      '#@@ServerRoot@@#$(prefix)#', '#@@ServerRoot@@'"##{opt_prefix}#"
+
+    args = %W[
+      --enable-mods-shared=all
+      --enable-unique-id
+      --enable-ssl
+      --enable-dav
+      --enable-cache
+      --enable-logio
+      --enable-deflate
+      --enable-cgi
+      --enable-cgid
+      --enable-suexec
+      --enable-rewrite
+      --with-apr=#{Formula["apr"].opt_prefix}
+      --with-apr-util=#{Formula["apr-util"].opt_prefix}
+      --with-pcre=#{Formula["pcre"].opt_prefix}
+      --with-ssl=#{Formula["openssl"].opt_prefix}
+      --with-port=8080
+      --with-sslport=8443
+      --with-mpm=event
+      --prefix=#{prefix}
+      --mandir=#{man}
+      --sysconfdir=#{etc}/apache2
+      --datadir=#{var}/www
+      --localstatedir=#{var}/apache2
```

Comment:
> For compatibility with other formulae and to keep with the naming convention, --libexecdir=#{libexec} should be added (previously in the config.layout).
> There isn't a configure option for installbuilddir from the layout file. By default, the Apache layout will set it as #{var}/www/build. This probably isn't a good location for these files as they will include cellar references. 

What is installed there?

Reply:
In libexec (modules in the current layout), the loadable shared modules are installed. In installbuilddir, are the files necessary for building additional shared modules later through apxs.

Reply:
@kabel What's the default location of `installbuilddir`?

Reply:
> What's the default location of installbuilddir?

`#{var}/www/build`

Reply:
Any concern about the other directories that don't have configure flags? They custom layout file seemed much more practical for keeping these organized.

Reply:
We should not need to override a bunch of defaults with some big heredoc table. That is ridiculous.

Reply:
All the distros seem to lay these directories out differently and unfortunately the default layout doesn't work well with how homebrew organizes things other formulae. Can you explain the ridiculousness?

Reply:
Configure should be sufficient as it is everywhere else.

Reply:
_Should be_, but I've given specific example here of where it is _not_.

Reply:
Then we'll need to live with the default.

Reply:
That's fine with me. I'm just trying to make people aware of potential support issues. Folks get upset when they can't find things where somebody told them it should be. 😉

Reply:
@kabel FYI `var` is defined as `HOMEBREW_PREFIX/"var"` (i.e., `/usr/local/var`) not `prefix/"var"` so I'm not sure which Cellar references you're talking about.

Reply:
Yes, I understand that. What I was mentioning was the files that are installed at `var/www/build/` will contain references to cellar files. For example, `/usr/local/var/www/build/config_vars.mk`.

Reply:
Those look benign because as far as I can tell they're not pointing at any user data, and they should be updated to the latest Cellar locations on upgrades.

Reply:
Hmm I think they won't be updated at all if there is a preexisting `/usr/local/var/www/build/config_vars.mk`, so they'd need to be updated in `post_install` and/or converted to opt paths.

But how does the heredoc from httpd24 help with this?
```
  def httpd_layout
    <<-EOS.undent
      <Layout Homebrew>
          prefix:        #{prefix}
          exec_prefix:   ${prefix}
          bindir:        ${exec_prefix}/bin
          sbindir:       ${exec_prefix}/bin
          libdir:        ${exec_prefix}/lib
          libexecdir:    ${exec_prefix}/libexec
          mandir:        #{man}
          sysconfdir:    #{etc}/apache2/2.4
          datadir:       #{var}/www
          installbuilddir: ${prefix}/build
          errordir:      ${datadir}/error
          iconsdir:      ${datadir}/icons
          htdocsdir:     ${datadir}/htdocs
          manualdir:     ${datadir}/manual
          cgidir:        #{var}/apache2/cgi-bin
          includedir:    ${prefix}/include/httpd
          localstatedir: #{var}/apache2
          runtimedir:    #{var}/run/apache2
          logfiledir:    #{var}/log/apache2
          proxycachedir: ${localstatedir}/proxy
      </Layout>
    EOS
```
Those all still look like Cellar paths, just as is in ` /usr/local/var/www/build/config_vars.mk` for the current iteration of the PR.

```
iMac-TMP:~ joe$ grep Cellar /usr/local/var/www/build/config_vars.mk 
exp_exec_prefix = /usr/local/Cellar/apache-httpd/2.4.27
exp_bindir = /usr/local/Cellar/apache-httpd/2.4.27/bin
exp_sbindir = /usr/local/Cellar/apache-httpd/2.4.27/bin
exp_libdir = /usr/local/Cellar/apache-httpd/2.4.27/lib
exp_libexecdir = /usr/local/Cellar/apache-httpd/2.4.27/libexec
exp_mandir = /usr/local/Cellar/apache-httpd/2.4.27/share/man
exp_includedir = /usr/local/Cellar/apache-httpd/2.4.27/include
PCRE_LIBS = -L/usr/local/Cellar/pcre/8.41/lib -lpcre
mandir = /usr/local/Cellar/apache-httpd/2.4.27/share/man
libexecdir = /usr/local/Cellar/apache-httpd/2.4.27/libexec
prefix = /usr/local/Cellar/apache-httpd/2.4.27
EXTRA_INCLUDES = -I$(includedir) -I. -I/usr/local/opt/apr/libexec/include/apr-1 -I/usr/local/opt/apr-util/libexec/include/apr-1 -I/usr/local/opt/openssl/include -I/usr/local/Cellar/pcre/8.41/include
```

Reply:
Oh I see, it's the reverse. You want to move `installbuilddir` to `${prefix}/build`. That seems like an odd location for something that by default would be exposed in a subdirectory of the datadir `#{var}/www`.

But yes we should replace the Cellar paths with opt paths in `config_vars.mk` if we do decide to use the default `#{var}/www/build` location.

Are there others besides installbuilddir you're concerned about?